packages/statemanager/src/stateManager.ts

```diff
@@ -139,9 +249,15 @@ export class DefaultStateManager extends BaseStateManager implements StateManage
     // @ts-expect-error
     await this._trie._db.put(key, value)
 
+    const keyHex = key.toString('hex')
+    this._codeCache[keyHex] = value
+
     if (this.DEBUG) {
       this._debug(`Update codeHash (-> ${short(codeHash)}) for account ${address}`)
     }
+    if (!(await this.getAccount(address))) {
```

Comment:
```suggestion
    if ((await this.getAccount(address) === undefined)) {
```

I'm surprised the linter isn't complaining here about an explicit boolean check.

Reply:
Wrong parantheses with await, online code suggestions are always a bit dangerous 😋.

But will generally update, thanks, still not sure about this linter thing.